### PR TITLE
basic alerting, for internal use primarily

### PR DESCRIPTION
**Pull request by heavycrystal**
For now, an alerting function logs an entry in an alerts table in catalog, and also dispatches notifications to one or more slack channels. Events are staggered to be 15 minutes apart at least, as long as the same "alert key" is used.

Only logs alerts for slot lag increasing a threshold, configured via the environment variable `PEERDB_SLOT_LAG_MB_ALERT_THRESHOLD`

### CODE REVIEW

File: flow/connectors/postgres/postgres.go [hunk 1]
```diff
@@ -79,6 +81,11 @@ func NewPostgresConnector(ctx context.Context, pgConfig *protos.PostgresConfig)
 		metadataSchema = *pgConfig.MetadataSchema
 	}
 
+	vigil, err := evervigil.NewVigil()
```
Comment (serprex) [inline flow/connectors/postgres/postgres.go:84]: should `NewVigil` take ctx given it's calling `Context.Background()`?

File: flow/utils/evervigil/ever_vigil.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+			notifier.UseServices(slackService)
+		default:
+			return fmt.Errorf("unknown service type: %s", serviceType)
+		}
+		return nil
+	})
+
+	return &EverVigil{
+		notifier:    notifier,
+		catalogPool: catalogPool,
```
Comment (serprex) [inline flow/utils/evervigil/ever_vigil.go:63]: If this is only happening every 15 minutes is there a point to keeping open a connection? Pool will keep it open. Could reconnect every 15 minutes

**Comment by iskakaushik** [comment]
@heavycrystal  could you limit this PR to just implement the alerting service for slot growth. Lets split out the temporal activity failure alerts.

File: flow/utils/evervigil/ever_vigil.go (hunk 2 above)
Comment (serprex) [reply flow/utils/evervigil/ever_vigil.go:72]: Yes, but that connection stays open afterwards during these 15 minute intervals between polling

Comment (serprex) [reply flow/utils/evervigil/ever_vigil.go:72]: The point of a pool is to keep connections open to share between callers to avoid connection handshake overhead, some links:
1. https://github.com/jackc/pgx/issues/494
2. https://github.com/jackc/pgx/issues/205
3. `MaxConnIdleTime` defaults to 30 minutes https://github.com/jackc/pgx/blob/ba05097642a802eb5a76343f27a9a3515ed86498/pgxpool/pool.go#L134

File: flow/utils/evervigil/ever_vigil.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	catalogPool *pgxpool.Pool
+}
+
+type slackServiceConfig struct {
+	AuthToken  string   `json:"auth_token"`
+	ChannelIDs []string `json:"channel_ids"`
+}
+
+func NewVigil() (*EverVigil, error) {
+	catalogPool, catalogErr := catalog.GetCatalogConnectionPoolFromEnv()
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/utils/evervigil/ever_vigil.go]: I'm not sure if pools can be shared across workflows/activities due to the implicit serializing/de-serializing that happens. But worth investigating.

[the remark this answers is not in the document]
Comment (serprex) [reply flow/utils/evervigil/ever_vigil.go]: https://community.temporal.io/t/go-sdk-dependency-injection/6823

If you create pool before `RegisterActivity` you can pass pool in there

File: flow/connectors/postgres/postgres.go (hunk 1 above)
Comment (serprex) [reply flow/connectors/postgres/postgres.go]: No, if you want that level of isolation you'll want the monitor to be a separate container

File: flow/utils/evervigil/ever_vigil.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	return &EverVigil{
+		notifier:    notifier,
+		catalogPool: catalogPool,
+	}, nil
+}
+
+func (ev *EverVigil) Close() {
+	if ev.catalogPool != nil {
+		ev.catalogPool.Close()
+	}
```
Comment (serprex) [inline flow/utils/evervigil/ever_vigil.go:79]: Catalog pool should never be closed

File: flow/utils/evervigil/ever_vigil.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/nikoksr/notify"
+	"github.com/nikoksr/notify/service/slack"
+	"github.com/sirupsen/logrus"
```
Comment (serprex) [inline flow/utils/evervigil/ever_vigil.go:13]: Should prefer slog over logrus in prep for #764

File: flow/utils/evervigil/ever_vigil.go [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/nikoksr/notify"
```
Comment (serprex) [inline flow/utils/evervigil/ever_vigil.go:11]: This library doesn't seem to support emailing as a service, so why not skip it & use `github.com/slack-go/slack` directly?

File: flow/connectors/core.go [hunk 7]
```diff
@@ -43,9 +43,6 @@ type CDCPullConnector interface {
 	// PullFlowCleanup drops both the Postgres publication and replication slot, as a part of DROP MIRROR
 	PullFlowCleanup(jobName string) error
 
-	// SendWALHeartbeat allows for activity to progress restart_lsn on postgres.
```
Comment (serprex) [inline flow/connectors/core.go:46]: Is this being removed because it's unused? Should be split out to another PR

**Comment by iskakaushik** [comment]
please rename EverVigil to something like `AlertingService` or something more generic. I like the name, but it isn't obvious what it does.

File: flow/utils/evervigil/ever_vigil.go (hunk 6 above)
Comment (heavycrystal) [reply flow/utils/evervigil/ever_vigil.go:11]: it does support e-mail, but switched to `slack-go` anyway

**Comment by heavycrystal** [comment]
reworked in #866
